src/BlockTypes/Checkout.php

```diff
@@ -179,18 +179,42 @@ protected function hydrate_customer_payment_methods( AssetDataRegistry $data_reg
 		remove_filter( 'woocommerce_payment_methods_list_item', [ $this, 'include_token_id_with_payment_methods' ], 10, 2 );
 	}
 
+	/**
+	 * Removes HTML tags and its content.
+	 *
+	 * @param string $text Text to remove tags from.
+	 */
+	protected function strip_tags_content( $text ) {
+		return preg_replace( '@<(\w+)\b.*?>.*?</\1>@si', '', $text );
+	}
+
 	/**
 	 * Hydrate the checkout block with data from the API.
 	 *
 	 * @param AssetDataRegistry $data_registry Data registry instance.
 	 */
 	protected function hydrate_from_api( AssetDataRegistry $data_registry ) {
+		// Save previous notices so we can return them in a `notices` property to
+		// be used in the frontend.
+		$previous_notices = wc_get_notices( 'error' );
+		// Prevent carried notices from being converted to exceptions when getting
+		// the Cart and Checkout endpoints..
+		wc_clear_notices();
+
 		if ( ! $data_registry->exists( 'cartData' ) ) {
 			$data_registry->add( 'cartData', WC()->api->get_endpoint_data( '/wc/store/cart' ) );
 		}
 		if ( ! $data_registry->exists( 'checkoutData' ) ) {
 			add_filter( 'woocommerce_store_api_disable_nonce_check', '__return_true' );
-			$data_registry->add( 'checkoutData', WC()->api->get_endpoint_data( '/wc/store/checkout' ) );
+			$checkout_data = WC()->api->get_endpoint_data( '/wc/store/checkout' );
+			if ( is_array( $previous_notices ) && count( $previous_notices ) > 0 ) {
```

Comment:
> If we do this for checkout, we should do it for cart block too (and consider other cases).

The Cart block shouldn't be affected. The issue happened because Checkout is calling `convert_notices_to_exceptions`, that means any notice would prevent the Checkout block from rendering and instead an error would be shown. Cart doesn't call that method so it's not affected and it shows notices like other pages.